server/compose/service/record.go

```diff
@@ -1329,6 +1357,54 @@ func (svc record) DeleteByID(ctx context.Context, namespaceID, moduleID uint64,
 	return nil
 }
 
+func (svc record) UndeleteByID(ctx context.Context, namespaceID, moduleID, recordID uint64) (err error) {
+    var (
+        aProps = &recordActionProps{
+            namespace: &types.Namespace{ID: namespaceID},
+            module:    &types.Module{ID: moduleID},
+        }
+        
+        ns *types.Namespace
+        m  *types.Module
+        r  *types.Record
+    )
+
+    err = func() error {
+        if namespaceID == 0 {
+            return RecordErrInvalidNamespaceID()
+        }
+        if moduleID == 0 {
+            return RecordErrInvalidModuleID()
+        }
+
+        ns, m, err = loadModuleCombo(ctx, svc.store, namespaceID, moduleID)
+        if err != nil {
+            return err
+        }
+
+        aProps.setNamespace(ns)
+        aProps.setModule(m)
+
+        return nil
+    }()
+
+    if err != nil {
+        return svc.recordAction(ctx, aProps, RecordActionUndelete, err)
+    }
+
+    r, err = svc.undelete(ctx, namespaceID, moduleID, recordID)
+    if err != nil {
+        return svc.recordAction(ctx, aProps, RecordActionUndelete, err)
+    }
+    aProps.setRecord(r)
+
+    if err = dalutils.ComposeRecordUpdate(ctx, svc.dal, m, r); err != nil {
+        return err
```

Comment:
Why not have this whole block in the lambda above? Wouldn't it make shorter with les record action calls?